File: src/mem/ruby/protocol/chi/ep/HnPersistenceBridge.py
from m5.params import NULL, Param
from m5.SimObject import SimObject


class HnPersistenceBridge(SimObject):
    type = "HnPersistenceBridge"
    cxx_header = "mem/ruby/protocol/chi/ep/HnPersistenceBridge.hh"
    cxx_class = "gem5::ruby::HnPersistenceBridge"

    ep_backend = Param.EPBackend(NULL, "Owning project EPBackend")

File: src/mem/ruby/protocol/chi/ep/HnPersistenceBridge.cc
#include "mem/ruby/protocol/chi/ep/HnPersistenceBridge.hh"

#include "mem/ruby/protocol/chi/ep/EPBackend.hh"

namespace gem5::ruby
{

HnPersistenceBridge::HnPersistenceBridge(const Params &p)
  : SimObject(p), _backend(p.ep_backend)
{
}

void
HnPersistenceBridge::registerHnPersistence(
    Addr linePa, Addr txnId, int sourceSocket, bool replacement, bool fullLine)
{
    // Generic CHI configurations install a disabled bridge. Only the project
    // UBCC configuration binds an EPBackend and needs persistence metadata.
    if (!_backend)
        return;
    _backend->registerHnPersistence(
        linePa, txnId, sourceSocket, replacement, fullLine);
}

} // namespace gem5::ruby
